libs/workspace/src/schematics/dt-component-example/index.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { strings } from '@angular-devkit/core';
+import {
+  apply,
+  chain,
+  mergeWith,
+  Rule,
+  template,
+  Tree,
+  url,
+  noop,
+} from '@angular-devkit/schematics';
+import { formatFiles } from '@nrwl/workspace';
+import { DtComponentExampleOptions } from './schema';
+import { normalize } from '@angular-devkit/core';
+import { LICENSE_HEADER } from '../../utils/common-utils';
+import { join } from 'path';
+import { getSourceFile, findNodes } from '../utils/ast-utils';
+import * as ts from 'typescript';
+import { commitChanges, InsertChange } from '../utils/change';
+
+interface DtExampleExtendedOptions {
+  componentSelector: string;
+  examplesModule: string;
+  exampleRoute: string;
+  selector: string;
+  componentModule: {
+    name: string;
+    package: string;
+  };
+  exampleComponent: {
+    component: string;
+    module: string;
+    modulesConstant: string;
+    template: string;
+  };
+  name: string;
+  component: string;
+}
+
+function generateComponentOptions(
+  options: DtComponentExampleOptions,
+): { name: string; package: string } {
+  return {
+    name: `Dt${strings.classify(options.component)}Module`,
+    package: `@dynatrace/barista-components/${strings.dasherize(
+      options.component,
+    )}`,
+  };
+}
+
+function generateExampleComponentOptions(
+  options: DtComponentExampleOptions,
+): {
+  component: string;
+  module: string;
+  modulesConstant: string;
+  template: string;
+} {
+  return {
+    component: `DtExample${strings.classify(options.name)}${strings.classify(
```

Comment:
Maybe store the classified name in a variable and re-use it would make code more readable

Reply:
I have taken a look and I don't think we use it as many places, so I would leave it like it is. I case of the dasherized strings I completely agree.